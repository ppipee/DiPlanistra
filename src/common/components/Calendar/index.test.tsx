import React from 'react'

import 'jest-styled-components'
import { render } from '@testing-library/react'

import * as CalendarType from '.'

describe('Test Calendar', () => {
	jest.doMock('./components/CalendarHeader', () => () => <div>CalendarHeader</div>)
	jest.doMock('./components/CalendarTable', () => () => <div>CalendarTable</div>)
	jest.doMock('./components/CalendarProvider', () => (props: any) => <div>{props.children}</div>)

	const { default: Calendar } = require('./index') as typeof CalendarType
	const setDate = () => {}

	it('should render Calendar with default date', () => {
		const MOCK_DEFAULT = {
			startDate: new Date(2020, 1),
			endDate: new Date(2020, 2),
		}
		const { container } = render(<Calendar setDate={setDate} defaultDate={MOCK_DEFAULT} />)

		expect(container.firstChild).toMatchSnapshot()
	})

	it('should render Calendar with out default date', () => {
		const { container } = render(<Calendar setDate={setDate}>children</Calendar>)

		expect(container.firstChild).toMatchSnapshot()
	})
})
