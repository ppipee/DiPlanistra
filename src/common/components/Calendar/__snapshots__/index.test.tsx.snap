// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Calendar should render Calendar with default date 1`] = `
.c0 {
  padding: 4px;
  display: inline-block;
  width: 280px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<div>
  <div
    class="c0"
  >
    <div>
      CalendarHeader
    </div>
    <div>
      CalendarTable
    </div>
  </div>
</div>
`;

exports[`Test Calendar should render Calendar with out default date 1`] = `
.c0 {
  padding: 4px;
  display: inline-block;
  width: 280px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<div>
  <div
    class="c0"
  >
    <div>
      CalendarHeader
    </div>
    <div>
      CalendarTable
    </div>
    children
  </div>
</div>
`;
